external_libraries/amgcl/adapter/zero_copy.hpp

```diff
@@ -52,7 +52,7 @@ zero_copy(size_t n, const Ptr *ptr, const Col *col, const Val *val) {
     auto A = std::make_shared< backend::crs<Val> >();
     A->nrows = n;
     A->ncols = n;
-    A->nnz   = ptr[n];
+    A->nnz   = n ? ptr[n] : 0;
```

Comment:
this seems like a delicate change
you sure abt this? Is it from upstream?

Reply:
This is also from upstream, but it does not include the recent fix for Kratos integration:
https://github.com/ddemidov/amgcl/compare/3a207307f9b..4f260881c715